apps/vaccines-updates-client/src/components/main-form/index.tsx

```diff
@@ -89,23 +92,35 @@ export const MainForm: FunctionComponent<{ className?: string }> = props => {
     const sendReport = useSendReport();
     const [formState, setFormState] = useState<FormState>('idle');
 
+    const { canSendReport, availableReportsToEdit, setAvailableReportsToEdit, setReportIdToEdit } = useFormData();
+
     const onSendClicked = useCallback(async () => {
         try {
-            setFormState('sending')
-            await sendReport();
-            setFormState('sent')
+            setFormState('sending');
+            const report = await sendReport();
+            setFormState('sent');
+
+            setReportIdToEdit(NewReport);
+            if (report.id) {
+                const index = availableReportsToEdit.reports.findIndex(r =>
+                    (r as { report: VaccinesReport }).report?.id?.pKey === report.id.pKey &&
+                    (r as { report: VaccinesReport }).report?.id?.internalId === report.id.internalId);
+                availableReportsToEdit.reports[index] = { report };
+            } else {
+                availableReportsToEdit.reports.push({ report });
+            }
```

Comment:
This looks like a great function to extract from there. Maybe `createOrUpdateReport`?

Reply:
Done.